app/scripts/modules/core/modal/buttons/modalClose.component.ts

```diff
@@ -1,17 +1,31 @@
-import {module} from 'angular';
+import {Component, Output, EventEmitter} from '@angular/core';
 
-class ModalCloseComponent implements ng.IComponentOptions {
-  public bindings: any = {
-    dismiss: '&'
-  };
-  public template = `
+import {IDowngradeItem} from 'core/domain/IDowngradeItem';
+@Component({
+  selector: 'deck-modal-close',
+  template: `
     <div class="close-button pull-right">
-      <a href class="btn btn-link" ng-click="$ctrl.dismiss()" >
+      <a href class="btn btn-link" (click)="close($event)" >
         <span class="glyphicon glyphicon-remove"></span>
       </a>
     </div>
-`;
+`
+})
+export class ModalCloseComponent {
+
+  @Output()
+  public dismiss = new EventEmitter();
+
+  public close($event: ng.IAngularEvent) {
+    $event.preventDefault();
+    this.dismiss.emit();
+  }
 }
 
 export const MODAL_CLOSE_COMPONENT = 'spinnaker.core.modal.modalClose.directive';
-module(MODAL_CLOSE_COMPONENT, []).component('modalClose', new ModalCloseComponent());
+export const MODAL_CLOSE_COMPONENT_DOWNGRADE: IDowngradeItem = {
+  moduleName: MODAL_CLOSE_COMPONENT,
+  injectionName: 'modalClose',
+  moduleClass: ModalCloseComponent,
+  outputs: ['dismiss']
```

Comment:
Would be nice to have a comment in place on things that are required for upgrade that might not be well documented, and then gives us a reference point for cruft we can rip out once we can do a full ng2 cutover...